packages/dx-scheduler-core/src/plugins/drag-drop-provider/helpers.ts

```diff
@@ -63,46 +65,84 @@ export const autoScroll: PureComputed<
   }
 };
 
-export const calculateAppointmentTimeBoundaries: CalculateAppointmentTimeBoundaries = (
+export const timeBoundariesByResize: TimeBoundariesByResize = (
+  payload, targetData, targetType, cellDurationMinutes, insidePart,
+) => {
+  if (targetType !== payload.appointmentType) {
+    return { appointmentStartTime: undefined, appointmentEndTime: undefined };
+  }
+
+  let appointmentStartTime;
+  let appointmentEndTime;
+  const sourceType = payload.type;
+
+  if (sourceType === RESIZE_TOP) {
+    const insideTopOffset = targetType === VERTICAL_TYPE
+      ? insidePart * cellDurationMinutes * 60 / 2 : 0;
+    appointmentStartTime = moment(targetData.startDate as Date)
+      .add(insideTopOffset, SECONDS).toDate();
+    appointmentEndTime = moment(payload.endDate as Date).toDate();
```

Comment:
Isn't `payload.endDate` a `Date` instance?
Can't it be like that?
```js
appointmentEndTime = payload.endDate;
```

Reply:
I prefer to create a new instance of Date. I want to change like that

```js
appointmentEndTime = new Date(payload.endDate);
```

Reply:
What is wrong with the old instance? Is it not always a `Date` or what?

The line means "we take appointment end as-is" or does it mean something more?